plugins/woocommerce/includes/admin/class-wc-admin-addons.php

```diff
@@ -1556,4 +1544,55 @@ private static function set_locale_data_in_transient( $transient, $value, $local
 		$transient_value[ $locale ] = $value;
 		return set_transient( $transient, $transient_value, $expiration );
 	}
+
+	/**
+	 * Make wp_safe_remote_get request to Woo.com endpoint.
+	 * Optionally pass user auth token, locale or country.
+	 *
+	 * @param string $url
+	 * @param ?array $options     Options for the request. For example, to pass auth token,
+	 *                            locale and country:
+	 *                            array(
+	 *                            'auth' => true,
+	 *                            'locale' => true,
+	 *                            'country' => true,
+	 *                            )
+	 *
+	 * @return array|WP_Error
+	 */
+	public static function fetch( $url, $options = array() ) {
+		$headers = array();
+
+		if ( $options['auth'] ) {
+			$auth = WC_Helper_Options::get( 'auth' );
+
+			if ( ! empty( $auth['access_token'] ) ) {
+				$headers['Authorization'] = 'Bearer ' . $auth['access_token'];
+			}
+		}
+
+		$parameters = array();
+		$parameter_string = '';
+
+		if ( $options['locale'] ) {
+			$parameters[] = http_build_query( array( 'locale' => get_user_locale() ) );
+		}
+
+		if ( $options['country'] ) {
+			$country = WC()->countries->get_base_country();
+			if ( ! empty( $country ) ) {
+				$parameters[] = http_build_query( array( 'country' => $country ) );
+			}
+		}
+
+		if ( ! empty( $parameters ) ) {
+			$parameter_string = '?' . implode( '&', $parameters );
+		}
+
+		return wp_safe_remote_get( $url . $parameter_string,
+			array(
+				'headers'    => $headers,
+				'user-agent' => 'WooCommerce/' . WC()->version . '; ' . get_bloginfo( 'url' ),
+			) );
+	}
```

Comment:
Maybe this can be simplified with something like so:

```php
    if ( ! empty( $options[ 'locale' ] ) ) {
        $parameters[ 'locale' ] = get_user_locale();
    }

    if ( ! empty($options['country'] ) ) {
        $country = WC()->countries->get_base_country();
        if ( ! empty( $country ) ) {
            $parameters[ 'country' ] = $country;
        }
    }

    $parameter_string = ! empty( $parameters ) ? '?' . http_build_query( $parameters ) : '';

    return wp_safe_remote_get( $url . $parameter_string, array(
        'headers'    => $headers,
        'user-agent' => 'WooCommerce/' . WC()->version . '; ' . get_bloginfo( 'url' ),
    ) );
    ```

Reply:
Beautiful! Changed to this in c8fc24668b79740d10263a092edef91da68c3dbf.